frb_codegen/src/generator/rust/mod.rs

```diff
@@ -162,7 +155,7 @@ impl Generator {
         lines.push(self.section_header_comment("executor"));
         lines.push(self.generate_executor(ir_file));
 
-        if self.options.should_generate_sync_execution_mode_utility {
+        if block_index == 1 {
```

Comment:
Not changed yet?

Reply:
> Not changed yet?

What exactly you want here?  Making it `if block_index == 0` ?   Do you also mean that when outputting more than 1 same name Api, the suffix should started from 0? like `new_uint_8_list_0`?

Reply:
```
#[derive(Copy)]
pub struct BlockIndex(usize);

impl BlockIndex {
  const PRIMARY = BlockIndex(0);
}

// when want to pass block_index:
fn some_function(block_index: BlockIndex) {}

// when want to say it is the primary one
// replace "if block_index == 1 {" with:
if block_index == BlockIndex::PRIMARY {

Reply:
> ```
> #[derive(Copy)]
> pub struct BlockIndex(usize);
> 
> impl BlockIndex {
>   const PRIMARY = BlockIndex(0);
> }
> 
> // when want to pass block_index:
> fn some_function(block_index: BlockIndex) {}
> 
> // when want to say it is the primary one
> // replace "if block_index == 1 {" with:
> if block_index == BlockIndex::PRIMARY {
> ```

Fixed in [cc7cfbc](https://github.com/fzyzcjy/flutter_rust_bridge/pull/527/commits/cc7cfbc4f1a0b944bff5d25383e8f87edc1a914f).